File: uncategorized/2073_time_needed_to_buy_tickets.ts
function timeRequiredToBuy(tickets: number[], k: number): number {

    let ans = 0;

    for(let [i,v] of tickets.entries()){
        if(i <= k){
            ans += Math.min(v,tickets[k]);
        } else {
            ans += Math.min(v,tickets[k]-1);
        }
    }

    return ans;

};

File: uncategorized/2073_time_needed_to_buy_tickets.rs
impl Solution {
    pub fn time_required_to_buy(tickets: Vec<i32>, k: i32) -> i32 {

        let mut ans:i32 = 0;
        let k = k as usize;

        for (i,v) in tickets.iter().enumerate() {
            if i <= k {
                ans += i32::min(*v, tickets[k]);
            } else {
                ans += i32::min(*v, tickets[k]-1);
            }
        }

        ans
    }
}